### PR TITLE
Adding FTP prediction models with power curve as input. 

### PR DESCRIPTION
**Pull request by adarshsivaram45**

### CODE REVIEW

**Review by MorgaineBarter — COMMENTED** [review]
this looks good, 
would like a bit more discussion on the results (in terms of what this accuracy means in the context of the prediction)

**Comment by adarshsivaram45** [comment]
> this looks good,
> would like a bit more discussion on the results (in terms of what this accuracy means in the context of the prediction)


Thanks for the feedback. Given below is the explanation for the above question: 

The accuracy metrics that we used - Mean Squared Error (MSE) and R² Score indicate how well our models predict the Functional Threshold Power (FTP) from the cycling data using power curves as input. In the context of FTP prediction, the results can be explained in the following way

Mean Squared Error (MSE)
- HistGradientBoostingRegressor: 6.6573
- Decision Tree: 0.0113
- Random Forest: 2.7128

MSE measures the average of the squares of the errors, i.e., the average squared difference between the predicted and actual FTP values. A lower MSE indicates that the predicted values are close to the actual values.

R² Score
- HistGradientBoostingRegressor: 0.9618
- Decision Tree: 0.9999
- Random Forest: 0.9844

R² Score, or the coefficient of determination, measures the proportion of variance in the dependent variable (FTP) that is predictable from the independent variable (power curve). An R² score close to 1 indicates that a large proportion of the variance in FTP is explained by the model.


Contextual Implications
- Decision Tree Model: Despite its high R² score and low MSE, the extremely high accuracy might indicate overfitting, where the model fits the training data very well but may not generalize as well to unseen data.
- HistGradientBoostingRegressor and Random Forest Models: Both models show strong predictive performance with high R² scores and relatively low MSEs, suggesting they are robust and likely generalize well to new data.

**Review by MorgaineBarter — APPROVED** [review]
thanks for the comments, next time add this contextual analysis to the results please.

**Comment by MorgaineBarter** [comment]
(the submission)
